MarketPlace/MarketPlace/Models/GoodsCatalog.cs

```
@@ -0,0 +1,60 @@
+﻿using MarketPlace.Interfaces;
+using System.Collections.Concurrent;
+
+namespace MarketPlace.Models
+{
+    public class GoodsCatalog : BaseEntity<int>, IGoodsCatalog
+    {
+        private readonly ConcurrentDictionary<int, Good> goodsCatalog = new();
+        private readonly object _syncObj = new();
+
+        public ConcurrentDictionary<int, Good> Goods { get => goodsCatalog; }
+
+        public void Create(Good entity)
+        {
+            if (entity != null)
```

Если делаете валидацию, то лучше эксепшн сразу выкидывать, а не игнорировать null. То есть:
if (entity == null) throw new ArgumentNullException(nameof(entity));
Принцип Fail Fast.

---

Валидацию в слое бизнес логике имеется в виду.